### PR TITLE
Displaying Historical Information on a user view

### LINKED ISSUE #300: Implement historical information from view
This will allow for the historical information of a user to be displayed on a page. For example, how many games a user has won, and achievements of that degree.

### PR DESCRIPTION
**Pull request by AndrewM131**
This pull request addresses issue #300 where it aims to display the correct historical information such as games won and such to a view where any view can access it.

### CODE REVIEW

**Comment by AndrewM131** [comment]
Update: I implemented the code such that there is statistics for matches played, matches won, tournaments played, tournaments won. I added the attributes in views.py in the users folder. I also went into userprofile_detail.html and added a link to the user's stats. The button works and goes to the user's stats. This can be accessed by any user if they go on the user's profile.

File: src/templates/users/userprofile_detail.html
Comment (24raniwalar) [inline src/templates/users/userprofile_detail.html]: Once you merge dev, you will see that the Edit Profile and Account buttons are sectioned off separately from the general profile buttons such as the Friend Request buttons and the Stats button. The Friend Request and stats buttons should be wrapped such that they either appear alongside the other buttons or such that there is space between the two rows of buttons.

**Review by 24raniwalar — CHANGES_REQUESTED** [review]
Changes to views and user_history align. The logic for displaying user stats in the user_history page works well, but the button placement is a little odd in userprofile_detail.html. I have added specific comments there to address the button and where it should be placed; other than that this PR LGTM!

File: src/templates/users/userprofile_detail.html
Comment (AndrewM131) [reply src/templates/users/userprofile_detail.html]: Got it, I'll add them into the if statements just like I did with the account button so it looks cleaner

**Review by 24raniwalar — APPROVED** [review]
LGTM! I think the if/else logic in the userprofile_detail.html page can be abbreviated since Stats shows up for all users, but that can be left for future developers to clean up in my opinion. Nice work!

File: src/chigame/users/views.py
Comment (elizabethli31) [inline src/chigame/users/views.py]: Hm this isn't quite right because it's just counting if the player is currently in a tournament (could be on going). If you could add another filter where the tournament_end_date < timezone.now() that should make this work.

File: src/chigame/users/views.py
Comment (elizabethli31) [inline src/chigame/users/views.py]: Same comment as for tournaments. I think you might want to use Lobby here instead of match and filter for match_status = 3.

**Review by elizabethli31 — CHANGES_REQUESTED** [review]
Thanks for adding back the account info and the stats page display looks really nice! A couple changes related to the way matches and tournaments played is being counted right now.

File: src/chigame/users/views.py
Comment (AndrewM131) [reply src/chigame/users/views.py]: That's a good filter I wasn't aware of, I'll use that

File: src/chigame/users/views.py
Comment (AndrewM131) [reply src/chigame/users/views.py]: I'll implement it since it has a match_status I can take advantage of

**Comment by AndrewM131** [comment]
I implemented the requested changes, but my only comment is that a user can be in a match if assigned by an admin, and not be in a lobby. I think there should be some functionality in the future to require a lobby for a match just so a person can't have matches won, but no matches played due to it being tied to lobbies.

**Comment by 24raniwalar** [comment]
Added back remove friend functionality; got removed elsewhere in another PR.

**Review by elizabethli31 — CHANGES_REQUESTED** [review]
Remove tournaments related files (just to avoid any accidental merge issues), and I'll approve.

**Review by elizabethli31 — APPROVED** [review]
Looks good now! Good work fixing up the matches/tournaments so they are taking in the proper info

**Comment by AndrewM131** [comment]
Closing comment: In this PR, we created a view that can be accessed in the profile of a user that displays the stats of a user, so for example, how many matches a user has been in, how many matches they've won, how many tournaments they've participated in and how many have they won. This is visible to any user when looking at another user's profile. One thing is that match count is linked to lobby, and since an admin can manually add a user to a match without a lobby, there should be something in the future that mandates a lobby for a match such that the numbers stay consistent. Also in the future, there should be more stats, such as rank in Chigame website, and maybe win rate, and things like that.
Team contributions can be found in issue closing statement.

**Comment by elizabethli31** [comment]
Issue Score: Excellent

Comments:
Great work using models from other teams to complete this PR!

**Pull request merged**
